MakieCore/src/conversion.jl

```diff
@@ -1,5 +1,19 @@
 
 function convert_arguments end
+
+"""
+    convert_attribute(value, attribute::Key[, plottype::Key])
+
+Convert `value` into into a suitable domain for use as `attribute`.
+
+# Example
+```jldoctest
+julia> using Makie
+
+julia> Makie.convert_attribute(:black, key"color"())
+RGBA{Float32}(0.0f0,0.0f0,0.0f0,1.0f0)
+```
+"""
 function convert_attribute end
```

Comment:
I have a couple of questions about this function. Is this an internal function? Is the user supposed to overload it? If so are there typical cases, when a user might want to overload this?

Reply:
For instance I might have a custom recipe `particles` that takes attributes
`photoncolor` and `electroncolor.
Should I overload 
```julia
MakieCore.convert_attributes(value, ::key"photoncolor", ::key"phsp") = convert_attributes(value, ::key"color")
MakieCore.convert_attributes(value, ::key"electroncolor", ::key"phsp") = convert_attributes(value, ::key"color")
```
?

Reply:
No, this is only for atomic plot types (like scatter, mesh, etc) and gets called in the backend.
You can use `to_color` in your recipe for such cases...
Overloading is only useful, if you have your own custom color type and want to use it directly as a color attribute!